api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java

```diff
@@ -84,17 +88,21 @@ public BiometricSubject enroll(BiometricSubject subject) {
 
     @Override
     public BiometricSubject update(BiometricSubject subject) {
-        M2SysCaptureResponse capture = scanDoubleFingers();
+        M2SysCaptureResponse fingerScan = scanDoubleFingers();
+        FingerScanStatus fingerScanStatus = checkIfFingerScanExists(fingerScan);
 
-        String response = localBioServerClient.update(subject.getSubjectId(), capture.getTemplateData());
-        M2SysResults results = XmlResultUtil.parse(response);
+        updateService.updateLocally(subject, fingerScan);
 
-        if (!results.isUpdateSuccess()) {
-            throw new M2SysBiometricsException("Unable to update fingerprints for: "
-                    + subject.getSubjectId());
+        String nationalId;
+        if (fingerScanStatus.isRegisteredNationally()) {
```

Comment:
We should also check if national client is correctly configured

Reply:
right